jogando uma excessao para ser pega em outro bloco de programa.



Models --> Classes


public class ExemploExcessao
    {
        public void Metodo1()
        {
            Metodo2();
        }

        public void Metodo2()
        {
            Metodo3();
        }

        public void Metodo3()
        {
            Metodo4();
        }

        public void Metodo4()
        {
            throw new Exception("Ocorreu uma excessão");
        }




Program.cs 


new ExemploExcessao().Metodo1();



Unhandled exception. System.Exception: Ocorreu uma excessão
   at ExemploExplorando.Models.ExemploExcessao.Metodo4() in C:\Users\João Pedro\Documents\estudos\.NET com C#\ExemploExplorando\Models\ExemploExcessao.cs:line 27    
   at ExemploExplorando.Models.ExemploExcessao.Metodo3() in C:\Users\João Pedro\Documents\estudos\.NET com C#\ExemploExplorando\Models\ExemploExcessao.cs:line 22    
   at ExemploExplorando.Models.ExemploExcessao.Metodo2() in C:\Users\João Pedro\Documents\estudos\.NET com C#\ExemploExplorando\Models\ExemploExcessao.cs:line 17    
   at ExemploExplorando.Models.ExemploExcessao.Metodo1() in C:\Users\João Pedro\Documents\estudos\.NET com C#\ExemploExplorando\Models\ExemploExcessao.cs:line 12    
   at Program.<Main>$(String[] args) in C:\Users\João Pedro\Documents\estudos\.NET com C#\ExemploExplorando\Program.cs:line 5